### PR TITLE
clean up the tmp Cert data

### PR DESCRIPTION
**Pull request by guangxuli**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33566)

### CODE REVIEW

File: pkg/kubectl/cmd/config/config_test.go [hunk 1]
```diff
@@ -294,7 +294,7 @@ func TestEmbedNoKeyOrCertDisallowed(t *testing.T) {
 
 func TestEmptyTokenAndCertAllowed(t *testing.T) {
 	fakeCertFile, _ := ioutil.TempFile("", "cert-file")
-
+	defer os.RemoveAll(fakeCertFile.Name())
```
Comment (justinsb) [inline pkg/kubectl/cmd/config/config_test.go]: I believe os.RemoveAll removes a directory, so os.Remove is the more normal choice.  But it turns out os.RemoveAll calls os.Remove first anyway, so ...

**Review by justinsb — CHANGES_REQUESTED** [review]
Suggest using os.Remove - unless there's a reason to use os.RemoveAll?

**Comment by guangxuli** [comment]
@justinsb thanks, yes, I made a mistake, I use the wrong function. I have modified the code.

**Review by justinsb — APPROVED** [review]
Thanks for fixing!

**Comment by justinsb** [comment]
@k8s-bot ok to test

**Comment by foxish** [comment]
@janetkuo Looks the jenkins bot didn't start one of the required tests and the bot continued to try and trigger it in every cycle (10 mins or so.). 
@fejta, is this a case that the mungegithub bot should take care of?

**Comment by janetkuo** [comment]
@k8s-bot gke e2e test this #33361

**Comment by justinsb** [comment]
@k8s-bot gke e2e test this

**Pull request merged**